docs/authentication_architecture.md

```diff
@@ -0,0 +1,77 @@
+to think if we want to use the token to tag the signaling messages (seems unnecessary), or simply `peer_id`, it's harder to manage permissions
+(more of a signaling architecture problems, but worth noting),
+- **BE** doesn't need to pass the token to **CL** (has to respond to in anyway to start signaling messages flow), but have to take care of matching incoming messages with `peer_id`s.~
+
+## Who should create tokens?
+
+### **BE**
+
+If **BE** (server SDK) is responsible for creating tokens, then it has to know the secret that will be used in signing JWT. It also needs `peer_id`, which can be obtained from **JF**. That takes care of **CL** authorization, but we also need to authorize **BE** - **JF** connection. One possible solution is to use the very same secret as before to create tokens that will be used by **BE**. So **BE** will generate token and then, instead of sending them to client, will use it itself (which may seem wierd, but we came to conclusion that the approach is alright).
+
+This approach only makes sense when using direct signaling, otherwise tokens are not necessary at all (except for **BE** - **JF** connection).
+
+### **JF**
+
+In this approach **JF** creates the tokens (when using `add_peer`, token is send in the response), so user only needs to pass it to the **CL**.
+You also might need to pass expected token presmissions to **JF**.
+Despite that, we still need a way to authenticate **BE** - **JF** connection. Possible solutions:
+
+- create JWT on **BE** side anyway (in such case you might as well use the first approach in order not to split logic responsible for token generation between Server SDK and Jellyfish),
+- create JWT (or some other token type) once and use it in configuration (makes it easier to change **BE** permissions, if that's ever necessary, but the token never expires, I'm not sure whether that's a problem, also **BE** doesn't need to know the secret).
+
+No matter who generates the tokens, effort from the user comes to passing the token to **CL** (in direct signaling, otherwise no need to do anything except for creating the signalling connection).
+
+## Consclusion
+
+We will be using **JF** to generate tokens, as it makes it easier to maintain (we don't need to implement token generation in all of our SDKs) and we keep all of the logic responsible for generation and validation together.
+Also, **BE** authentication might not be very difficult: they may just share a common secret and use it directly to authenticate, especially when **JF** and **BE** are in the same internal network, which will be a common case.
+Using **BE** to generate tokens could also complicate things when using more than one instance of **JF** and make it harder to scale and make modifications related to tokens (need to redeploy both **JF** and **BE** in such case).
```

Comment:
Could we elaborate this? IMO, the only difference is that we either use bare secret for authentication or create some token using this secret. In both cases we need some secret so I am not sure how the second approach (when generating token on BE) affects scenario with multiple JF